core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -75,13 +81,68 @@ private[spark] class IndexShuffleBlockResolver(conf: SparkConf) extends ShuffleB
   }
 
   /**
+   * Check whether there are index file and data file also they are matched with each other, returns
+   * the lengths of each block in data file, if there are matched, or return null.
+   */
+  private def checkIndexAndDataFile(index: File, data: File, blocks: Int): Array[Long] = {
+    val lengths = new Array[Long](blocks)
+    if (index.length() == (blocks + 1) * 8) {
```

Comment:
instead of the nested `if-else`'s, it might be clearer to write this as:
```
// can you add a comment here to explain this check
if (index.length() != (blocks + 1) * 8) {
  return null
}
```